docs/code/VBufferCareFeeding.md

```diff
@@ -142,129 +136,130 @@ using (IRowCursor cursor = dataview.GetRowCursor(col => col == 5))
+`VBuffer<T>`.) The editor is past that line considered to "own" the internal
+structure of the `VBuffer<T>` you passed in. (So, correct code should not
+continue to use that `VBuffer<T>` structure from that point onwards.)
+
+Then, one parties on the `Span<T>` for values and, if in a sparse application,
+the `Span<int>` for indices, on that editor, in whatever way is appropriate
+for the task.
+
+Last, one calls one of the `Commit` methods to get another `VBuffer`, with the
+values and indices accessible through the `VBuffer<T>` methods the same as
+those that had been set in the editor structure.
+
+Similar to how creating a `VBufferEditor<T>` out of a `VBuffer<T>` renders the
+passed in `VBuffer<T>` invalid, likewise, getting the `VBuffer<T>` out of the
+editor out of the `VBufferEditor<T>` through one of the commit methods renders
+the *editor* invalid. In both cases, "ownership" of the internal buffers is
+passed along to the successor structure, rendering the original structure
+invalid, in some sense.
+
+Internally, these buffers are backed by arrays that are reallocated *if
+needed* by the editor upon its creation, but reused if they are large enough.
+
+## Ownership of `VBuffer`
+
+Nonetheless, this does not mean that bugs are impossible, because the concept
+of buffer reuse does carry the implication that we must be quite clear about
+who "owns" a `VBuffer` at any given time, for which we have developed this
+convention:
+
+Unless clearly specified in documentation, a caller is assumed to always own a
+`VBuffer` as returned by `ref`. This means they can do whatever they like to
+it, like pass the same variable into some other getter, or modify its values.
 Indeed, this is quite common: normalizers in ML.NET get values from their
 source, then immediately scale the contents of `Values` appropriately. This
 would hardly be possible if the callee was considered to have some stake in
 that result.
 
+The `ValueGetter` indicated above is the most important example of this
+principle. By passing in an existing `VBuffer<T>` into that delegate by
+reference, they are giving the implementation control over it to use (or,
+reallocate) as necessary to store the resulting value. But, once the delegate
+returns with the value returned, the caller is now considered to own that
+buffer.
+
 There is a corollary on this point: because the caller owns any `VBuffer`,
-then you shouldn't do anything that irrevocably destroys their usefulness to
+then could should not do anything that irrevocably destroys its usefulness to
 the caller. For example, consider this method that takes a vector `src`, and
 stores the scaled result in `dst`.
 
 ```csharp
-VectorUtils.ScaleBy(ref VBuffer<float> src, ref VBuffer<float> dst, float c)
+ScaleBy(in VBuffer<float> src, ref VBuffer<float> dst, float c)
 ```
 
 What this does is, copy the values from `src` to `dst`, while scaling each
-value seen by `c`.
+value from `src` seen by the factor `c`.
```

Comment:
``value from `src` seen by the factor `c`.``

=>

``value in `src` by the factor `c`.``

?

Reply:
Well... that's not right now, is it though? We *don't* scale each value in `src` by `c`. In fact the entire point of the paragraph following that is an explanation of why that is precisely the wrong thing to do. I can try to rephrase to try to make it a bit more direct and awkwardly phrased, but I would not welcome the specific edit suggested.